File: V2/client/swingui/src/main/resources/script/robot.js
importPackage(java.lang) ;
importPackage(com.neverwinterdp.swing.robot) ;

function Robot() {
  this.units = [] ;
  
  this.add = function(name, description, method) {
    var unit = {name: name, description: description, method: method } ;
    this.units.push(unit) ;
  };
  
  this.run = function() {
    for(var i = 0; i < this.units.length; i++) {
      var unit = this.units[i] ;
      unit.status = '' ;
    }
    var frameui = new FrameUI(uiroot) ;
    for(var i = 0; i < this.units.length; i++) {
      var unit = this.units[i] ;
      unit.method(frameui) ;
      unit.status = 'OK' ;
      Thread.sleep(500) ;
    }
    frameui.destroy() ;
  };
  
  this.report = function() {
    var header = ["Name", "Status"] ;
    var width  = [30,     10] ;
    console.println("") ;
    console.println('-----------------------------------------') ;
    console.println(header, width) ;
    console.println('-----------------------------------------') ;
    for(var i = 0; i < this.units.length; i++) {
      var unit = this.units[i] ;
      console.println([unit.name, unit.status], width) ;
    }
  };
}

File: V2/client/swingui/src/main/java/com/neverwinterdp/swing/console/robot/Robot.java
package com.neverwinterdp.swing.console.robot;

import java.awt.Frame;
import java.util.ArrayList;
import java.util.List;

public class Robot {
  private Frame frame ; 
  private List<Action> actions = new ArrayList<Action>() ;
  private long animationSpeed = 1000 ;
  
  public Robot(Frame frame) {
    this.frame = frame ;
  }
  
  public long getAnimationSpeed() { return this.animationSpeed ; }
  public void setAnimationSpeed(long speed) { this.animationSpeed = speed ; }
  
  public Robot add(Action action) {
    actions.add(action) ;
    return this ;
  }
  
  public void run() throws Exception {
    FrameUI frameui = new FrameUI(frame) ;
    try {
      for(int i = 0; i < actions.size(); i++) {
        Thread.sleep(animationSpeed) ;
        Action action = actions.get(i) ;
        action.execute(frameui) ;
      }
    } finally {
      frameui.destroy() ;
    }
  }
  
  public void destroy() {
  }
}